morecantile/models.py

```diff
@@ -1147,12 +1185,16 @@ def xy_bounds(self, *tile: Tile) -> BoundingBox:
         )
 
         left = origin_x + math.floor(t.x / cf) * matrix.cellSize * cf * matrix.tileWidth
-        top = origin_y - t.y * matrix.cellSize * matrix.tileHeight
         right = (
             origin_x
             + (math.floor(t.x / cf) + 1) * matrix.cellSize * cf * matrix.tileWidth
         )
-        bottom = origin_y - (t.y + 1) * matrix.cellSize * matrix.tileHeight
+        if matrix.cornerOfOrigin == "topLeft":
+            top = origin_y - t.y * matrix.cellSize * matrix.tileHeight
+            bottom = origin_y - (t.y + 1) * matrix.cellSize * matrix.tileHeight
+        else:
+            bottom = origin_y + t.y * matrix.cellSize * matrix.tileHeight
+            top = origin_y + (t.y + 1) * matrix.cellSize * matrix.tileHeight
```

Comment:
The "mirror_factor" (`mf`) could be also used here: 

```python
top = origin_y - mf * (t.y + int(mf < 0)) * t.y * matrix.cellSize * matrix.tileHeight
bottom = origin_y - mf * (t.y + int(mf > 0)) * t.y * matrix.cellSize * matrix.tileHeight
```

Reply:
thanks for the comments @claxn 🙏 

I kinda like the verbosity of the code because it gives a clearer view of what is going on. But I'll sleep on this tonight and re-assess tomorrow morning :-)